### PR TITLE
feat(scoring): Add a scoring mechanism

### PR DESCRIPTION
**Pull request by sam-mathews**
This pull request introduces a basic scoring mechanism and enhances various game assets:

1. Implemented a new scoring system:
   - Game starts with a preset bank balance
   - Balance decreases when purchasing food for fish
   - Prevents feeding fish if there are insufficient funds
   - Collecting coins from fish increases the bank balance
   - Added `score.c` and `score.h` to manage this functionality

2. Enhanced sprite and tile management:
   - Added new sprite generator utility in Go (`utils/spritegen`)
   - Generated new number and symbol tiles for displaying score
   - Updated background initialisation to include new tiles

3. Improved game object interactions:
   - Modified coin and cursor logic to work with the new scoring system
   - Updated fish behaviour to use proper random number generation

4. Refactored and cleaned up existing code:
   - Applied consistent code formatting
   - Improved code organisation and readability

5. Updated build system:
   - Modified `Makefile` to include new asset generation steps

These changes provide a more engaging gameplay experience by introducing a monetary system and improving visual elements. The new sprite generator utility also allows for easier asset management in the future.

### CODE REVIEW

File: src/score.c
Comment (LuminousSpike) [inline src/score.c:1]: The use of the string drawing here means we're running out of VRAM.
image
image

This is how it should look like normally
image
image

What we'll need to do is either handle drawing strings ourselves with our own bitmap font, or look into some other way.

**Review by LuminousSpike — CHANGES_REQUESTED** [review]
Just needs the VRAM corruption issue sorted out.

**Review by LuminousSpike — APPROVED** [review]
The graphical issues have been resolved. Awesome work! Approved.

**Comment by sam-mathews** [comment]
Will need some more tidying, but good to get a stick in the ground.

**Pull request merged**